### DISCUSSION TITLE
`output-file` now doesn't take a path

### DISCUSSION THREAD

**Discussion by Worville** [quarto-dev/quarto-cli#4058] [Q&A] [answered]
I've built a report in Quarto with the following parameters:

```format: 
  pdf:
    documentclass: report
    classoption: landscape, a4paper
    papersize: A4
    geometry:
      - top=20mm
      - bottom=20mm
      - left=5mm
      - right=5mm
      - heightrounded
    output-file: "reports/report_name.pdf"
```

When I last ran this report (Dec 2021) there was no problem. Now I'm receiving an error:

```ERROR: 
In file report.qmd
  Invalid value for `output-file`: paths are not allowed
```

What is the right approach now to define both an output filename and an output directory for the report to be created in?

**Comment by cderv** [comment, marked as answer]
Yes there was a changed in v1.2 due to this issue: https://github.com/quarto-dev/quarto-cli/issues/2266
`output-file` needs to be a file name

To control output directory for your rendering, you could leverage Quarto projects  which have a `output-dir` configuration. (https://quarto.org/docs/reference/projects/core.html).

**Reply by Worville** [reply]
Thanks a lot for the quick reply! Will take a look.

**Reply by Worville** [reply]
Works perfectly, thanks again.

**Comment by cdugeai** [comment]
Is it possible to have different output dirs for each file of the project as we cannot specify it in ` output-file` ?

**Reply by mcanouil** [reply]
Use `output-dir` but only for Quarto projects, see https://quarto.org/docs/projects/profiles.html#profile-configuration

- https://github.com/quarto-dev/quarto-cli/discussions/2171

Alternatively, use post-render scripts to copy/move contents after rendering.
